neo4j/db/errors.go

```diff
@@ -23,12 +23,34 @@ import (
 	"strings"
 )
 
+type ErrorClassification string
+
+const (
+	ClientError    ErrorClassification = "CLIENT_ERROR"
+	DatabaseError  ErrorClassification = "DATABASE_ERROR"
+	TransientError ErrorClassification = "TRANSIENT_ERROR"
+	UnknownError   ErrorClassification = "UNKNOWN"
+)
+
 // Neo4jError is created when the database server failed to fulfill request.
+// TODO 6.0: rename to GqlError
 type Neo4jError struct {
-	Code           string
-	Msg            string
+	// TODO 6.0: to be deprecated.
+	Code      string
+	Msg       string
+	GqlStatus string
+	// TODO 6.0: rename to StatusDescription
+	GqlStatusDescription string
+	// TODO 6.0: rename to Classification
+	GqlClassification ErrorClassification
+	// TODO 6.0: rename to RawClassification
+	GqlRawClassification string
+	// TODO 6.0: rename to DiagnosticRecord
+	GqlDiagnosticRecord map[string]any
+	// TODO 6.0: rename to Cause
+	GqlCause       *Neo4jError
```

Comment:
Some docs for those fields would be nice. Have a look at the ADR and/or other drivers for inspiration.

Reply:
Docs added 👍